feat: add Speaker in FOSS Chapter Event, Online Option in Event

Praneel7015 opened the pull request:
## Status

- [ ] Draft
- [x] Ready for review

---

## Related Issue

Closes / Addresses: #<issue_number>


---

## Problem

<!-- Briefly describe the problem or motivation for this PR -->
1) There is no clear way to add speakers for an event in any FOSS Club Event without the use of CFP.
2) Hard to Distinguish an Online Event from an Offline Event in Event Type

---

## Solution

<!-- Describe your solution and what you changed -->
1) Worked With @idlip to Understand the Code, and Added the Tag
2) Added an Online option such that event Host-er will be able to set event type to online event

---

## Progress (All must be checked to merge)

- [x] Tested locally (Needs more checking)
- [ ]  

---

## Changelog

<commit>: <!-- e.g., feat: add user login endpoint -->

feat:  add Speaker in Role Selection Option and add Online Option Selection in Event type.

---

## Visualization

<!-- Add screenshots, diagrams, or screen recordings if applicable -->
<img width="1598" height="826" alt="image" src="https://github.com/user-attachments/assets/81f6a93f-1a95-4df0-a1fb-539424f18084" />
*Added Online Option*

<img width="1595" height="822" alt="image" src="https://github.com/user-attachments/assets/5d37f401-d10c-4c4d-b1b2-61a4e9f8516e" />
*Added Speaker Option*

<img width="1598" height="825" alt="image" src="https://github.com/user-attachments/assets/c2ca3fb1-ca9c-4dac-9d4d-222cb0349659" />
*Volunteers and Speakers*
---

## Further Ahead

<!-- Mention any follow-up work or related PRs planned -->
Just Learnt How to Use Frappe and the repo expect mistakes 😓 
Let me know if any issues arise

Expect more soon! 

 Future :
-  Dockerize Entire Setup For Ease of Setup.
- Extensive FAQ for Setup and Contributing to the repo.
- Contributing to the Documentation


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Event Type now includes "Online".
  * Event member roles now include "Speaker".
  * Event member record layout set to "Dynamic".

* **Documentation**
  * Added header/license comments to event code.

* **Chores**
  * Added/updated development tooling configuration for pre-commit.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Praneel7015 committed:
add: Speaker in select

Praneel7015 committed:
fix: linting fixes

Praneel7015 committed:
Merge branch 'fossunited:develop' into speaker

idlip commented:
Hi praneel, thanks for the PR.

So tldr; it adds "speaker" role to foss event member table and "Online" type for event.

We can remove event_speakers table?
Since its same as event_members?

I'm not sure if its unintuitive to add speaker as event_member with speaker role via dashboard, but that actually gets our job done, if not we may have to add section in dashboard to add speakers.

I think since its for Online mostly, we can simply inform people to add speaker with role to "Volunteers" in dashboard?

<img width="1312" height="609" alt="image" src="https://github.com/user-attachments/assets/3d20593e-6eae-4b40-9895-ebfc4aa5a299" />


This PR is fine to add changes to backend now.
Next PR can address the frontend issue.

We've get_speakers function that needs to append any speaker role from this table, and match it with `volunteers_card.html` macro file

Praneel7015 committed:
Merge branch 'fossunited:develop' into speaker

Praneel7015 committed:
fix: remove speaker table in foss_chapter_event

Praneel7015 committed:
fix: Remove Speaker tab in Tickets Section of foss_chapter_event

Praneel7015 committed:
Remove duplicate noqa comments in imports

Praneel7015 committed:
remove # noqa : E501 at lines 73 and 74 of foss_chapter_event.py

Praneel7015 commented:
Removed the Redundant Event_Speaker table present in the volunteers section.
Removed the Mistakenly placed Speaker Table in Tickets page
Removed duplicate `# noqa: E501` directive that appears twice on lines 39 and 42 of  file `foss_chapter_event.py`
Removed  `# noqa: E501` directive that appears on lines 73,74 of file `foss_chapter_event.py`

idlip commented:
Yep this is good to go now.

Thanks alot for taking this up!
